File: launch_school/RB101-Foundations/small_problems/daily_double.rb
def crunch(str)
  char_arr = ''
  str.each_char do |char|
    char_arr << char unless char_arr[-1] == char
  end
  char_arr
end

puts crunch('ddaaiillyy ddoouubbllee') == 'daily double'
puts crunch('4444abcabccba') == '4abcabcba'
puts crunch('ggggggggggggggg') == 'g'
puts crunch('a') == 'a'
puts crunch('') == ''

File: launch_school/JS210-Fundamentals-of-Javascript/small_problems/easy2/daily_double.js
function crunch(string) {
  let result = '';

  for (let char of string) {
    let lastChar = result[result.length - 1];
    if (char !== lastChar) {
      result += char;
    }
  }

  console.log(result);
  return result;
}

crunch('ddaaiillyy ddoouubbllee');    // "daily double"
crunch('4444abcabccba');              // "4abcabcba"
crunch('ggggggggggggggg');            // "g"
crunch('a');                          // "a"
crunch('');                           // ""
